File: sdk/python/examples/list_events.py
"""Example: list recent events for a contract."""

import os

from soroban_pulse import SorobanPulseClient

client = SorobanPulseClient(api_key=os.environ["SOROBAN_PULSE_API_KEY"])

for event in client.iter_events(contract_id="CABC123", limit=50):
    print(event["id"], event["event_type"], event["ledger"])

File: sdk/javascript/examples/list-events.ts
import { SorobanPulseClient } from "../src";

async function main() {
  const client = new SorobanPulseClient({ apiKey: process.env.SOROBAN_PULSE_API_KEY! });

  for await (const event of client.iterEvents({ contractId: "CABC123", limit: 50 })) {
    console.log(event.id, event.eventType, event.ledger);
  }
}

main().catch(console.error);
